server/src/main/java/io/crate/planner/operators/NestedLoopJoin.java

```diff
@@ -395,21 +396,63 @@ private Tuple<Collection<String>, List<MergePhase>> configureExecution(Execution
             nlExecutionNodes = leftResultDesc.nodeIds();
             left.setDistributionInfo(DistributionInfo.DEFAULT_SAME_NODE);
             right.setDistributionInfo(DistributionInfo.DEFAULT_BROADCAST);
-            rightMerge = JoinOperations.buildMergePhaseForJoin(plannerContext, rightResultDesc, nlExecutionNodes);
+            rightMerge = buildMergePhaseForJoin(plannerContext, rightResultDesc, nlExecutionNodes);
         } else {
             // run join phase non-distributed on the handler
             left.setDistributionInfo(DistributionInfo.DEFAULT_BROADCAST);
             right.setDistributionInfo(DistributionInfo.DEFAULT_BROADCAST);
-            if (JoinOperations.isMergePhaseNeeded(nlExecutionNodes, leftResultDesc, false)) {
-                leftMerge = JoinOperations.buildMergePhaseForJoin(plannerContext, leftResultDesc, nlExecutionNodes);
+            if (isMergePhaseNeeded(nlExecutionNodes, leftResultDesc, false)) {
+                leftMerge = buildMergePhaseForJoin(plannerContext, leftResultDesc, nlExecutionNodes);
             }
-            if (JoinOperations.isMergePhaseNeeded(nlExecutionNodes, rightResultDesc, false)) {
-                rightMerge = JoinOperations.buildMergePhaseForJoin(plannerContext, rightResultDesc, nlExecutionNodes);
+            if (isMergePhaseNeeded(nlExecutionNodes, rightResultDesc, false)) {
+                rightMerge = buildMergePhaseForJoin(plannerContext, rightResultDesc, nlExecutionNodes);
             }
         }
         return new Tuple<>(nlExecutionNodes, Arrays.asList(leftMerge, rightMerge));
     }
 
+    static MergePhase buildMergePhaseForJoin(PlannerContext plannerContext,
```

Comment:
imho, this is worse than having them in a common place and avoid `HashJoin` <-> `NestedLoopJoin` dependency.

Wouldn't be better to move them to `JoinPlanBuilder` instead?

Reply:
we could move them into `JoinPlan`.

Reply:
However, this has nothing really to do with LogicalPlans, it an helper to build an execution plan.

Reply:
`JoinPlan` is the interface for the rules and is also focused on LogicalPlan. Also does not really fit there.

Reply:
But it's in the `operators` package, where NestedLoop and HashJoin are, which they implement this iface.

Reply:
I think like this it's acceptable.

Reply:
I agree, converting JoinPlan to abstract class it's a better alternative.